packages/compiler-cli/src/ngtsc/typecheck/extended/checks/text_attribute_not_binding/index.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AST, TmplAstNode, TmplAstTextAttribute} from '@angular/compiler';
+import ts from 'typescript';
+
+import {ErrorCode, ExtendedTemplateDiagnosticName} from '../../../../diagnostics';
+import {NgTemplateDiagnostic} from '../../../api';
+import {TemplateCheckFactory, TemplateCheckWithVisitor, TemplateContext} from '../../api';
+
+/**
+ * Ensures that attributes that have the "special" angular binding prefix (attr., style., and
+ * class.) are interpreted as bindings. For example, `<div attr.id="my-id"></div>` will not
+ * interperet this as an `AttributeBinding` to `id` but rather just a `TmplAstTextAttribute`. This
+ * is likely not the intent of the developer. Instead, the intent is likely to have the `id` be set
+ * to 'my-id'.
+ */
+class TextAttributeNotBindingSpec extends
+    TemplateCheckWithVisitor<ErrorCode.TEXT_ATTRIBUTE_NOT_BINDING> {
+  override code = ErrorCode.TEXT_ATTRIBUTE_NOT_BINDING as const;
+
+  override visitNode(
+      ctx: TemplateContext<ErrorCode.TEXT_ATTRIBUTE_NOT_BINDING>,
+      component: ts.ClassDeclaration,
+      node: TmplAstNode|AST,
+      ): NgTemplateDiagnostic<ErrorCode.TEXT_ATTRIBUTE_NOT_BINDING>[] {
+    if (!(node instanceof TmplAstTextAttribute)) return [];
+
+    const name = node.name;
+    if ((!name.startsWith('attr.') && !name.startsWith('style.') && !name.startsWith('class.')) ||
+        !node.value) {
+      return [];
+    }
+
+    const boundSyntax = node.sourceSpan.toString();
+    let errorString: string;
+    if (name.startsWith('attr.')) {
+      const staticAttr = name.replace('attr.', '');
+      errorString = `Static attributes should be written without the 'attr.' prefix: ${
+          staticAttr}="${node.value}".`;
+
+    } else {
+      const expectedKey = `[${name}]`;
+      const expectedValue =
+          node.value === 'true' || node.value === 'false' ? node.value : `'${node.value}'`;
```

Comment:
nit: for better readability:
```suggestion
          (node.value === 'true' || node.value === 'false') ? node.value : `'${node.value}'`;
```

Reply:
Btw, could you plz also add a quick comment to describe why do we special-case `true` and `false` (but not numbers, `null`, `undefined`, etc)?